internal/subscription/validation.go

```diff
@@ -55,6 +55,8 @@ func getValidatedModel(ctx context.Context, req *metrov1.Subscription) (*Model,
 	m.Labels = req.GetLabels()
 	m.ExtractedSubscriptionProjectID = p
 	m.ExtractedSubscriptionName = s
+	m.RetryTopic = topic.GetTopicName(p, s+topic.RetryTopicSuffix)
```

Comment:
Would be good if we add a comment here explaining why we are using the subscription name and not the topic name to create retry and dlq topics.

Reply:
adding to the model